rmw_opendds_cpp/src/process_topic_and_service_names.cpp

```
@@ -23,54 +24,27 @@
 #include "rmw_opendds_shared_cpp/namespace_prefix.hpp"
 #include "rmw_opendds_shared_cpp/opendds_include.hpp"
 
-bool
-_process_topic_name(
+std::string
+get_topic_str(
   const char * topic_name,
-  bool avoid_ros_namespace_conventions,
-  char ** topic_str)
+  bool avoid_ros_namespace_conventions)
 {
-  bool success = true;
-  rcutils_allocator_t allocator = rcutils_get_default_allocator();
-  const char * topic_prefix = avoid_ros_namespace_conventions ? "" : ros_topic_prefix;
-  char * concat_str = rcutils_format_string(allocator, "%s%s", topic_prefix, topic_name);
-  if (concat_str) {
-    *topic_str = CORBA::string_dup(concat_str);
-    allocator.deallocate(concat_str, allocator.state);
-  } else {
-    RMW_SET_ERROR_MSG("could not allocate memory for topic string");
-    success = false;
+  if (!topic_name || strlen(topic_name) == 0) {
+    return "";
+  }
+  if (avoid_ros_namespace_conventions) {
+    return topic_name;
```

Since the topic_name is passed by pointer, I'm wondering if it wouldn't be better to return an error code instead.

---

This helper function is to decorate the topic_name. If it's null or empty, an empty string is returned so the caller can take an appropriate action (return RMW_RET_ERROR or nullptr).